Gordon360/ApiControllers/ScheduleControlController.cs

```diff
@@ -0,0 +1,196 @@
+using Gordon360.Repositories;
+using Gordon360.Models.ViewModels;
+using Gordon360.AuthorizationFilters;
+using Gordon360.Static.Names;
+using System;
+using Gordon360.Exceptions.ExceptionFilters;
+using Gordon360.Exceptions.CustomExceptions;
+using System.Collections.Generic;
+using System.Security.Claims;
+using System.Linq;
+
+namespace Gordon360.Controllers.Api
+{
+    [RoutePrefix("api/schedulecontrol")]
+    [CustomExceptionFilter]
+    [Authorize]
+    public class ScheduleControlController : ApiController
+    {
+        private IUnitOfWork _unitOfWork;
+
+        //declare services we are going to use.
+        private IProfileService _profileService;
+        private IAccountService _accountService;
+        private IRoleCheckingService _roleCheckingService;
+
+        private IScheduleControlService _scheduleControlService;
+
+
+        public ScheduleControlController()
+        {
+            _unitOfWork = new UnitOfWork();
+            _scheduleControlService = new ScheduleControlService(_unitOfWork);
+            _accountService = new AccountService(_unitOfWork);
+        }
+
+        public ScheduleControlController(IScheduleControlService scheduleControlService)
+        {
+            _scheduleControlService = scheduleControlService;
+        }
+
+        /// <summary>
+        /// Get schedule information of local user
+        /// </summary>
+        /// <returns></returns>
+        [HttpGet]
+        [Route("")]
+        public IHttpActionResult Get()
+        {
+            var authenticatedUser = this.ActionContext.RequestContext.Principal as ClaimsPrincipal;
+            var id = authenticatedUser.Claims.FirstOrDefault(x => x.Type == "id").Value;
+
+            object scheduleControlResult = null;
+            
+            scheduleControlResult = _unitOfWork.ScheduleControlRepository.GetById(id);
+
+            if (scheduleControlResult == null)
+            {
+                return NotFound();
+            }
+
+            return Ok(scheduleControlResult);
+
+        }
+
+
+
+        /// <summary>
+        /// Get schedule information of specific user
+        /// </summary>
+        /// <param name="username">username</param>
+        /// <returns></returns>
+        [HttpGet]
+        [Route("{username}")]
+        public IHttpActionResult Get(string username)
+        {
+            object scheduleControlResult = null;
+
+            var id = _accountService.GetAccountByUsername(username).GordonID;
+            scheduleControlResult = _unitOfWork.ScheduleControlRepository.GetById(id);
+            
+            if (scheduleControlResult == null)
+            {
+                return NotFound();
+            }
+
+            return Ok(scheduleControlResult);
+
+        }
+
+
+
+        /// <summary>
+        /// Update privacy of schedule
+        /// </summary>
+        /// <param name="value">Y or N</param>
+        /// <returns></returns>
+        [HttpPut]
+        [Route("update/privacy/{value}")]
```

Comment:
Do we need the word update in this route?

Reply:
Change made